File: leetcode_cpp/climbing_stairs.cpp
class Solution {
public:
    unordered_map <int, int> cached = {
        {0, 1},
        {1, 1}
    };

    int climbStairs(int n) {
        if (cached.count(n) == 0) {
            cached[n] = climbStairs(n - 1) + climbStairs(n - 2);
        }
        return cached[n];
    }
};

File: leetcode_python/climbing_stairs.py
class Solution:
    # @param n, an integer
    # @return an integer

    cached = {0: 1, 1: 1}

    def climbStairs(self, n):
        if n not in self.cached:
            self.cached[n] = self.climbStairs(n - 1) + self.climbStairs(n - 2)
        return self.cached[n]
